### PR TITLE
Move all verification tests to use shared test file

### PR DESCRIPTION
**Pull request by jhand2**
Make it easier to run DPE tests against different targets by putting all the test cases in a test vector file. This allows test infrastructure for a specific target to add required setup and transport code and then execute the tests in a loop.

In particular, this is to support easily running these tests against a Caliptra HwModel target.

Additional fixes:

* Make hex string conversion use all caps. This matches what Caliptra uses for subject/issuer serialNumber fields.
* Support encoding X.509 Directory Strings as either PrintableString or UTF8String. This allows more flexibility when specifying issuer encodings.

### CODE REVIEW

**Review by zhalvorsen — DISMISSED** [review]

File: verification/getCertificateChain.go [hunk 1]
```diff
@@ -14,19 +12,8 @@ import (
...
-		if err.Error() == "Requested support is not supported in the emulator" {
-			t.Skipf("[WARNING]: Failed executing TestGetCertificateChain command due to unsupported request. Hence, skipping the command execution")
-		} else {
-			log.Fatalf("[ERROR]: %s", err.Error())
-		}
-	}
-	testGetCertificateChain(instance, t)
+func TestGetCertificateChain(d TestDPEInstance, t *testing.T) {
+	testGetCertificateChain(d, t)
 }
```
[the remark this answers is not in the document]
Comment (jhand2) [reply verification/getCertificateChain.go:17]: Done

File: verification/tagTCI.go [hunk 2]
```diff
@@ -11,19 +11,8 @@ import (
...
-			t.Skipf("Warning: Failed executing TestTagTCI command due to unsupported request. Hence, skipping the command execution")
-		} else {
-			log.Fatal(err)
-		}
-	}
-
-	testtagTCI(instance, t)
+func TestTagTCI(d TestDPEInstance, t *testing.T) {
+	testtagTCI(d, t)
 }
```
[the remark this answers is not in the document]
Comment (jhand2) [reply verification/tagTCI.go]: Done

File: verification/verification_test.go [hunk 3]
```diff
@@ -3,128 +3,45 @@
...
-		instance.SetLocality(DPE_SIMULATOR_AUTO_INIT_LOCALITY)
-		return instance, nil
-	}
-	return nil, errors.New("Error in creating dpe instances - invalid test target type.")
-}
 
-// Get the emulator target
-func GetEmulatorTarget(support_needed []string) (TestDPEInstance, error) {
+			if !HasSupportNeeded(d, test.SupportNeeded) {
+				t.Skipf("Warning: Target does not have required support, skipping test.")
```
[the remark this answers is not in the document]
Comment (jhand2) [reply verification/verification_test.go:31]: Ya, I think that makes sense. Opened https://github.com/chipsalliance/caliptra-dpe/issues/225 to track.

File: ci.sh
Comment (zhalvorsen) [inline ci.sh:1]: Do you want to do the TODOs listed in here for the simulator profiles in this PR, or should we do it in a follow on?

**Review by zhalvorsen — APPROVED** [review]

**Review by zhalvorsen — APPROVED** [review]

**Pull request merged**